wp-includes/sqlite-ast/class-wp-sqlite-information-schema-reconstructor.php

```diff
@@ -0,0 +1,644 @@
+		$type = strtoupper( $column_type );
+
+		/*
+		 * Following the rules of column affinity:
+		 *   https://sqlite.org/datatype3.html#determination_of_column_affinity
+		 */
+
+		// 1. If the declared type contains the string "INT" then it is assigned
+		//    INTEGER affinity.
+		if ( str_contains( $type, 'INT' ) ) {
+			return 'int';
+		}
+
+		// 2. If the declared type of the column contains any of the strings
+		//    "CHAR", "CLOB", or "TEXT" then that column has TEXT affinity.
+		if ( str_contains( $type, 'TEXT' ) || str_contains( $type, 'CHAR' ) || str_contains( $type, 'CLOB' ) ) {
+			return 'text';
+		}
+
+		// 3. If the declared type for a column contains the string "BLOB" or
+		//    if no type is specified then the column has affinity BLOB.
+		if ( str_contains( $type, 'BLOB' ) || '' === $type ) {
+			return 'blob';
+		}
+
+		// 4. If the declared type for a column contains any of the strings
+		//    "REAL", "FLOA", or "DOUB" then the column has REAL affinity.
+		if ( str_contains( $type, 'REAL' ) || str_contains( $type, 'FLOA' ) ) {
+			return 'float';
+		}
+		if ( str_contains( $type, 'DOUB' ) ) {
+			return 'double';
+		}
+
+		/**
+		 * 5. Otherwise, the affinity is NUMERIC.
+		 *
+		 * While SQLite defaults to a NUMERIC column affinity, it's better to use
+		 * TEXT in this case, because numeric SQLite columns in non-strict tables
+		 * can contain any text data as well, when it is not a well-formed number.
+		 *
+		 * See: https://sqlite.org/datatype3.html#type_affinity
+		 */
+		return 'text';
+	}
+
+	/**
+	 * Escape a string literal for MySQL DEFAULT values.
+	 *
+	 * @param  string $literal The string literal to escape.
+	 * @return string          The escaped string literal.
+	 */
+	private function escape_mysql_string_literal( string $literal ): string {
+		// See: https://www.php.net/manual/en/mysqli.real-escape-string.php
+		return "'" . addcslashes( $literal, "\0\n\r'\"\Z" ) . "'";
+	}
+
+	/**
+	 * Quote an SQLite identifier.
+	 *
+	 * Wrap the identifier in backticks and escape backtick values within.
+	 *
+	 * @param  string $unquoted_identifier The unquoted identifier value.
+	 * @return string                      The quoted identifier value.
+	 */
+	private function quote_sqlite_identifier( string $unquoted_identifier ): string {
+		return '`' . str_replace( '`', '``', $unquoted_identifier ) . '`';
+	}
+
+	/**
+	 * Unquote a quoted MySQL identifier.
+	 *
+	 * Remove bounding quotes and replace escaped quotes with their values.
+	 *
+	 * @param  string $quoted_identifier The quoted identifier value.
+	 * @return string                    The unquoted identifier value.
+	 */
+	private function unquote_mysql_identifier( string $quoted_identifier ): string {
+		$first_byte = $quoted_identifier[0] ?? null;
+		if ( '"' === $first_byte || '`' === $first_byte ) {
+			$unquoted = substr( $quoted_identifier, 1, -1 );
+			return str_replace( $first_byte . $first_byte, $first_byte, $unquoted );
```

Comment:
Ditto my state machine question from earlier. Should we use the parser for this?